tests/foreman/destructive/test_capsulecontent.py

```diff
@@ -91,3 +92,97 @@ def test_positive_sync_without_deadlock(
 
     sync_status = large_capsule_configured.nailgun_capsule.content_sync(timeout='90m')
     assert sync_status['result'] == 'success', 'Capsule sync task failed.'
+
+
+@pytest.mark.tier4
+@pytest.mark.skip_if_not_set('capsule')
+def test_positive_sync_without_deadlock_after_rpm_trim_changelog(
+    target_sat, capsule_configured, function_entitlement_manifest_org
+):
+    """Promote a CV published with larger repos into multiple LCEs, assign LCEs to blank Capsule.
+    Assert that the sync task succeeds and no deadlock happens.
+
+    :id: 91c6eec9-a582-46ea-9898-bdcaebcea2f1
+
+    :setup:
+        1. Blank external capsule which is not synced yet, running with 4 & more pulpcore-workers
+
+    :steps:
+        1. Sync one bigger repositories to the Satellite.
+        2. Create a Content View, add the repository and publish it.
+        3. Promote it to 10+ LCEs and assign them to Capsule
+        4. Sync the Capsule
+        5. pulpcore-manager rpm-trim-changelogs --changelog-limit x
+        6. Sync Capsule again and verify no deadlocks occurs.
+        7. Repeat step 5 with a lesser changelog limit and step 6 again
+
+    :expectedresults:
+        1. Sync passes without deadlock.
+
+    :customerscenario: true
+
+    :BZ: 2170535, 2218661
+    """
+    org = function_entitlement_manifest_org
+    rh_repos = []
+    tasks = []
+    LCE_COUNT = 10
+    for name in ['rhel8_bos', 'rhel8_aps']:
+        rh_repo_id = target_sat.api_factory.enable_rhrepo_and_fetchid(
+            basearch=constants.DEFAULT_ARCHITECTURE,
+            org_id=org.id,
+            product=constants.REPOS[name]['product'],
+            repo=constants.REPOS[name]['name'],
+            reposet=constants.REPOS[name]['reposet'],
+            releasever=constants.REPOS[name]['releasever'],
+        )
+        # Sync step because repo is not synced by default
+        rh_repo = target_sat.api.Repository(id=rh_repo_id).read()
+        task = rh_repo.sync(synchronous=False)
+        tasks.append(task)
+        rh_repos.append(rh_repo)
+    for task in tasks:
+        target_sat.wait_for_tasks(
+            search_query=(f'id = {task["id"]}'),
+            poll_timeout=2500,
+        )
+        task_status = target_sat.api.ForemanTask(id=task['id']).poll()
+        assert task_status['result'] == 'success'
+
+    cv = target_sat.publish_content_view(org, rh_repos)
+    cvv = cv.version[0].read()
+
+    for i in range(LCE_COUNT):
+        lce = target_sat.api.LifecycleEnvironment(name=f'lce{i}', organization=org).create()
+        cvv.promote(data={'environment_ids': lce.id, 'force': False})
+        capsule_configured.nailgun_capsule.content_add_lifecycle_environment(
+            data={'environment_id': lce.id}
+        )
```

Comment:
I would recommend storing `capsule_configured.nailgun_capsule` in a local variable as nailgun_capsule is not a cached property and we are burning time on capsule search every time we call it.

Reply:
yes, I had a same thought about this, updated, thanks!